refactor(product): product 와 productLike 엔티티 재정의

Ko-udon opened the pull request:
## 변경사항

<!-- 이 PR에서 어떤점들이 변경되었는지 기술해주세요.  -->

1. .gitIgnore 에 관해 log 파일을 무시하도록 설정

2. product 엔티티 재정의

3. productLike 엔티티 추가


## 체크 리스트 (Checklist)

pull request 전에 아래 체크 리스트들을 만족하는 지 확인한 후 체크('x') 표시를 해주시기 바랍니다.

- [x]  master 브랜치가 아니라 **develop** 브랜치에 Merge하도록 pull request를 작성 중이신가요?
- [x]  모든 **테스트**가 성공했나요?

ehddn committed:
refactor(product): product 와 productLike 엔티티 재정의

1. .gitIgnore log 파일 설정

2. product 엔티티 재정의

3. productLike 엔티티 추가

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/product/model/entity/ProductEntity.java:
```diff
@@ -18,21 +23,28 @@
 @AuditOverride(forClass = BaseEntity.class)
 @Entity
-public class ProductEntity extends BaseEntity {
+public class ProductEntity extends BaseEntity  {
```
entity 패키지 안에 있어서 클래스 명에서 Entity는 빼도 되지 않을까요?

Ko-udon replied:
네 이부분은 반영해서 ProductEntity -> Product로 변경할게요.

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/product/model/entity/ProductEntity.java:
```diff
@@ -18,21 +23,28 @@
+    private Boolean isBest;
+
+    @OneToOne(mappedBy = "productEntity")
+    private ProductLike productLike;
```
ProductLike와 연관관계를 양방향으로 설정하신 이유가 뭘까요??

Ko-udon replied:
Like에서의 Product 접근이 필요할지 몰라 확신이 없는 상황에서 작성하다보니 그런것 같네요. Product에 대한 접근, 수정은 최소한으로 하기로 했으니 단방향이 맞는것 같습니다. 이부분도 수정할게요.

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/product/model/entity/ProductLike.java:
```diff
@@ -0,0 +1,34 @@
+
+    @Id
+    @Column(name = "prodcutId")
+    private Long productId;
```
이건 상품 ID가 아니라 productLikeId 같은데 이러면 아래 연관관계 JoinColumn 명도 동일하게 productId구요.
이렇게되면 이러면 나중에 햇갈릴거 같아요

Ko-udon replied:
erd설계를 보면 ProductLike는 본인의 Id를 별도로 가지는게 아닌, Product의 Id를 본인의 Id로 가지는 구조라 이렇게 작성했어요. 이 부분은 조금 더 생각해볼게요.

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/product/model/type/ProductCategory.java:
```diff
@@ -0,0 +1,7 @@
+package com.zerobase.wishmarket.domain.product.model.type;
+
+public enum ProductCategory {
+    CLOTHES, HOME_ELECTRONICS, TOY, IT_DEVICE, JEWELRY, FURNITURE, ETC
```
해당 카테고리당 맵핑되는 INT 값도 같이 작성하는 쪽으로 코드 수정이 이뤄지면 좋을거 같아요.
나중에 카테고리가 추가된다면 저렇게 나열만 되어 있으면 명확하게 알아보기 힘들 거 같네요.

위에서 INT 값도 같이 작성한다는 건 주석이 아니라 코드적인 부분입니다.

Ko-udon replied:
네 이부분은 반영하겠습니다.

ehddn committed:
refactor(product_modify): product, productLike 구조 변경

1. productEntity -> product 로 엔티티 명칭 수정

2. product -> productLike  일대일
단방향 구조로 변경

3. 상품의 카테고리별로 맵핑되는 INT값 부여

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/product/model/type/ProductCategory.java:
```diff
@@ -1,7 +1,14 @@
-    CLOTHES, HOME_ELECTRONICS, TOY, IT_DEVICE, JEWELRY, FURNITURE, ETC
+    CLOTHES(0), HOME_ELECTRONICS(1), TOY(2),
+    IT_DEVICE(3), JEWELRY(4), FURNITURE(5),
+    ETC(6)
```
Pre기획 쪽에 카테고리 번호를 이미 정해놨어요.
참고하셔서 수정해주세요~
